File: code/projectMQ/src/geterrorfromxml.py
# #coding=utf-8
# from xml.etree import ElementTree as ET
# per=ET.parse('errorcode.xml')
# p=per.findall('./login/item')

# for oneper in p:
#     for child in oneper.getchildren():
#         #print child.tag,':',child.text
#         print(child)


# # p=per.findall('./item')

# # for oneper in p:
# #     for child in oneper.getchildren():
# #         print child.tag,':',child.text

# import xml.etree.ElementTree as ET
# from docx import Document

# def parse_xml(file_path):
#     tree = ET.parse(file_path)
#     root = tree.getroot()
#     items = []
#     for item in root.findall('Error'):
#         title = item.find('error_code').text
#         description = item.find('error_description').text
#         items.append((title, description))
#     return items

# def create_word_document(items, output_path):
#     doc = Document()
#     for title, description in items:
#         doc.add_heading(title, level=1)
#         doc.add_paragraph(description)
#     doc.save(output_path)

# if __name__ == "__main__":
#     xml_file = '/opt/wupei/git/code/projectMQ/src/errorcode.xml'
#     output_docx = '/opt/wupei/git/code/projectMQ/src/output.docx'
    
#     items = parse_xml(xml_file)
#     create_word_document(items, output_docx)
#     print(f"Content has been imported into {output_docx}")




import xml.etree.ElementTree as ET
from docx import Document

def parse_xml(file_path):
    tree = ET.parse(file_path)
    root = tree.getroot()
    items = []
    for item in root.findall('item'):
        title = item.find('title').text
        description = item.find('description').text
        items.append((title, description))
    return items

def create_word_document(items, output_path):
    doc = Document()
    for title, description in items:
        doc.add_heading(title, level=1)
        doc.add_paragraph(description)
    doc.save(output_path)

if __name__ == "__main__":
    xml_file = '/opt/wupei/git/code/projectMQ/src/sample.xml'
    output_docx = '/opt/wupei/git/code/projectMQ/src/output.docx'
    
    items = parse_xml(xml_file)
    create_word_document(items, output_docx)
    print(f"Content has been imported into {output_docx}")

File: code/projectMQ/src/geterrorfromxml.cpp
// #include <iostream>
// #include <fstream>
// #include <DuckX.h>
// #include "tinyxml2.h"

// using namespace std;
// using namespace tinyxml2;

// void parseXML(const char *filename)
// {
//     XMLDocument doc;
//     doc.LoadFile(filename);

//     XMLElement *root = doc.RootElement();
//     XMLElement *error;
//     XMLElement *attribute;

//     // Create a new Word document
//     Document *document = new Document();
//     document->Create();

//     for (error = root->FirstChildElement("Error"); error; error = error->NextSiblingElement("Error"))
//     {
//         int index = atoi(error->Attribute("index"));

//         Paragraph *paragraph = document->AddParagraph();
//         paragraph->AddText("Error Index: " + to_string(index));

//         for (attribute = error->FirstChildElement("Attribute"); attribute; attribute = attribute->NextSiblingElement("Attribute"))
//         {
//             string name = attribute->Attribute("name");
//             string value = attribute->Attribute("value");

//             paragraph = document->AddParagraph();
//             paragraph->AddText(name + ": " + value);
//         }
//     }

//     // Save the document
//     document->Save("output.docx");
//     document->Close();
//     delete document;

//     cout << "Content has been imported into output.docx" << endl;
// }

// int main()
// {
//     parseXML("errors.xml");
//     return 0;
// }

#include <iostream>
#include <fstream>
#include <DuckX.h>
#include "tinyxml2.h"

using namespace std;
using namespace tinyxml2;

void parseXML(const char *filename)
{
    XMLDocument doc;
    doc.LoadFile(filename);

    XMLElement *root = doc.RootElement();
    XMLElement *error;
    XMLElement *attribute;

    // Create a new Word document
    Document *document = new Document();
    document->Create();

    for (error = root->FirstChildElement("Error"); error; error = error->NextSiblingElement("Error"))
    {
        int index = atoi(error->Attribute("index"));

        Paragraph *paragraph = document->AddParagraph();
        paragraph->AddText("Error Index: " + to_string(index));

        string errorCode, errorDescription, errorSolution;

        for (attribute = error->FirstChildElement("Attribute"); attribute; attribute = attribute->NextSiblingElement("Attribute"))
        {
            string name = attribute->Attribute("name");
            string value = attribute->Attribute("value");

            if (name == "error_code")
            {
                errorCode = value;
            }
            else if (name == "error_description")
            {
                errorDescription = value;
            }
            else if (name == "error_solution")
            {
                errorSolution = value;
            }
        }

        paragraph = document->AddParagraph();
        paragraph->AddText("Error Code: " + errorCode);
        paragraph = document->AddParagraph();
        paragraph->AddText("Error Description: " + errorDescription);
        paragraph = document->AddParagraph();
        paragraph->AddText("Error Solution: " + errorSolution);
    }

    // Save the document
    document->Save("output.docx");
    document->Close();
    delete document;

    cout << "Content has been imported into output.docx" << endl;
}

int main()
{
    parseXML("errors.xml");
    return 0;
}